python/pyspark/sql/tests.py

```diff
@@ -3367,6 +3367,33 @@ def test_ignore_column_of_all_nulls(self):
         finally:
             shutil.rmtree(path)
 
+    # SPARK-24721
+    def test_datasource_with_udf_filter_lit_input(self):
+        from pyspark.sql.functions import udf, lit, col
+
+        path = tempfile.mkdtemp()
+        shutil.rmtree(path)
+        try:
+            self.spark.range(1).write.mode("overwrite").format('csv').save(path)
+            filesource_df = self.spark.read.csv(path)
+            datasource_df = self.spark.read \
+                .format("org.apache.spark.sql.sources.SimpleScanSource") \
+                .option('from', 0).option('to', 1).load()
+            datasource_v2_df = self.spark.read \
+                .format("org.apache.spark.sql.sources.v2.SimpleDataSourceV2") \
```

Comment:
This wouldn't work if test classes are not compiled. I think we should better make another test suite that skips the test if the test classes are not existent.

Reply:
@HyukjinKwon I actually am not sure how does pyspark find these classes and how to check the existence, do you have an example?

Reply:
I can probably check try to check the existence of sql/core/target/scala-2.11/test-classes

Reply:
Hmm... I think this is a bit fragile because things like "scala-2.11"  (scala version can change).

Seems a bit over complicated to do this properly, when do we expect pyspark test to run without compiling scala test classes?

Reply:
Added checks to skip the tests if scala tests are not compiled